Assets/Scripts/UI/InGame/RegistrationCounter.cs

```diff
@@ -0,0 +1,30 @@
+﻿using UnityEngine;
+
+public sealed class RegistrationCounter
```

Comment:
I like to make my classes sealed when they weren't designed to be extended. If we wanted this to be extendable, we would add more `protected` and `virtual` members. Let me know which is preferred.
@Lithrun

Reply:
Sealed is fine, however, I found the naming of this class quite strange. "RegistrationCounter", so, we are Registering a Counter. The events also have strange names. It sounds like that you are trying to abstract counters, which perhaps future classes could also use? But if so, then why? Sounds like over-engineering to me.